scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -365,6 +367,31 @@ def test_gh_4511_regression(self):
         bounds = [(-5, 5), (-5, 5)]
         result = differential_evolution(rosen, bounds, popsize=1815, maxiter=1)
 
+    def test_calculate_population_energies(self):
+        # if popsize is 2 then the overall generation has size (4,)
+        solver = DifferentialEvolutionSolver(rosen, self.bounds, popsize=2)
+        solver._calculate_population_energies()
+
+        assert_equal(np.argmin(solver.population_energies), 0)
+
+        # initial calculation of the energies should require 4 nfev.
+        assert_equal(solver._nfev, 4)
+
+    def test_evolve_generator(self):
+        # if popsize is 2 then the overall generation has size (4,)
+        solver = DifferentialEvolutionSolver(rosen, self.bounds, popsize=2,
+                                             maxfun=8)
+        evolver = solver.evolve()
+        x, fun = next(evolver)
```

Comment:
I wonder if you couldn't just make `DifferentialEvolutionSolver` an iterator (i.e., implementing a `__next__` method). This way, you'd just do:
```
solver = DifferentialEvolutionSolver(rosen, self.bounds, popsize=2, maxfun=8)
x, fun = next(solver)
```
No need to expose this `evolve` method, which you have to initialize, etc.